### PR TITLE
wip: eval how to revamp

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: docs/evaluation/how_to_guides/datasets/export_filtered_traces_to_dataset.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 sidebar_position: 6
 ---
 
-# Export filtered traces from experiment to dataset
+# How to export filtered traces from experiment to dataset
```
Comment (agola11) [inline docs/evaluation/how_to_guides/datasets/export_filtered_traces_to_dataset.mdx:5]: I'm not sure we gain anything by putting How-to in every title? These guides are already in the how-to section

File: docs/evaluation/how_to_guides/evaluation/async.mdx
Comment (agola11) [inline docs/evaluation/how_to_guides/evaluation/async.mdx:1]: This one is going to be much needed

File: docs/evaluation/how_to_guides/evaluation/dataset_subset.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- [guide on fetching examples](../datasets/manage_datasets_programmatically#fetch-examples).
+- [guide on creating/managing dataset splits](../datasets/manage_datasets_in_application#create-and-manage-dataset-splits)
+
+:::
+
+# How to: Evaluate on a filtered view of a dataset
+
+You can use the `list_examples` / `listExamples` method to fetch a subset of examples from a dataset to evaluate on. You can refer to guide above to learn more about the different ways to fetch examples.
+
+One common workflow is to fetch examples that have a certain metadata key-value pair.
```
Comment (agola11) [inline docs/evaluation/how_to_guides/evaluation/dataset_subset.mdx:22]: would comment that we have much more powerful filtering capabilities (particularly for metadata) using the filter parameter: https://docs.smith.langchain.com/evaluation/how_to_guides/datasets/manage_datasets_programmatically#list-examples-by-structured-filter

File: docs/evaluation/how_to_guides/index.md [hunk 3]
```diff
@@ -1,63 +1,85 @@
...
+
+## Offline evaluation
+
+Evaluate and improve your application.
+
+### Run an evaluation
+- [Run an evaluation using the SDK](./how_to_guides/evaluation/evaluate_llm_application)
+- [Run an evaluation asynchronously](./how_to_guides/evaluation/async)
+- [Run an evaluation comparing two experiments](./how_to_guides/evaluation/evaluate_pairwise)
+- [Run an evaluation of a LangChain / LangGraph object](./how_to_guides/evaluation/langchain_runnable)
```
Comment (agola11) [inline docs/evaluation/how_to_guides/index.md:17]: Probably worth it to cover LangChain and LangGraph in separate guides?

There's some more interesting things in LG like evaluating a single node, traectory, etc. It might be worth it to call out the documentation for running a larger eval in the LG eval guide.

We have this tutorial I hacked together with Lance a while back but it's definitely overly complicated and I wonder if bits of it can be adapted to a more concise how to guide https://docs.smith.langchain.com/evaluation/tutorials/agents

**Review by agola11 — COMMENTED** [review]
some flyby comments but overall like the structure

File: docs/evaluation/how_to_guides/index.md [hunk 4]
```diff
@@ -1,63 +1,85 @@
...
+For comprehensive descriptions of every class and function see the [API Reference](https://langsmith-sdk.readthedocs.io/en/latest/evaluation.html).
+
+## Offline evaluation
+
+Evaluate and improve your application.
+
+### Run an evaluation
+- [Run an evaluation using the SDK](./how_to_guides/evaluation/evaluate_llm_application)
+- [Run an evaluation asynchronously](./how_to_guides/evaluation/async)
+- [Run an evaluation comparing two experiments](./how_to_guides/evaluation/evaluate_pairwise)
```
Comment (agola11) [inline docs/evaluation/how_to_guides/index.md:16]: nit: Run a comparative evaluation?

**Review by agola11 — APPROVED** [review]

File: docs/evaluation/how_to_guides/datasets/export_filtered_traces_to_dataset.mdx (hunk 1 above)
Comment (baskaryan) [reply docs/evaluation/how_to_guides/datasets/export_filtered_traces_to_dataset.mdx:5]: once you're on the page its not clear where you are bc sidebar doesn't expand that deep. personally like that langchain docs are all titled "How to..." but dont feel strongly

File: docs/evaluation/how_to_guides/index.md (hunk 3 above)
Comment (baskaryan) [reply docs/evaluation/how_to_guides/index.md:17]: i kinda think this should live in langgraph docs and be linked from here

File: docs/evaluation/how_to_guides/evaluation/evaluate_llm_application.mdx
Comment (baskaryan) [inline docs/evaluation/how_to_guides/evaluation/evaluate_llm_application.mdx:1]: @agola11  can review this one

File: docs/evaluation/how_to_guides/evaluation/evaluate_llm_application.mdx [hunk 5]
```diff
@@ -163,469 +150,220 @@ The values in the examples can also take any form, such as strings, numbers, lis
...
     typescript`
       import type { EvaluationResult } from "langsmith/evaluation";
       import type { Run, Example } from "langsmith/schemas";
       
-      // Row-level evaluator
-      function correctLabel(rootRun: Run, example: Example): EvaluationResult {
+      function correct(rootRun: Run, example: Example): EvaluationResult {
         const score = rootRun.outputs?.outputs === example.outputs?.output;
-        return { key: "correct_label", score };
+        return { key: "correct", score };
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/evaluate_llm_application.mdx:173]: Is this code correct? The `score` doesn't have a corresponding value....

File: docs/evaluation/how_to_guides/evaluation/langgraph.mdx [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+
+    # Finally, we compile it!
+    # This compiles it into a LangChain Runnable,
+    # meaning you can use it as you would any other runnable.
+    # Note that we're (optionally) passing the memory when compiling the graph
+    app = workflow.compile()
+
+`,
+    typescript`
+// ToDo
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/langgraph.mdx:97]: Do we need to fill out before merging in?

File: docs/evaluation/how_to_guides/evaluation/llm_as_judge.mdx [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import {
+  CodeTabs,
+  python,
+  typescript,
+} from "@site/src/components/InstructionsWithCode";
+
+# How to define an LLM-as-a-judge evaluator
+
+:::info Key concepts
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/llm_as_judge.mdx:9]: Shouldn't be part of this PR since it'll take too much time, but users want a lot more guidance on good LLM as judge definition than we give rn. Even our concepts doc isn't very helpful.

File: docs/evaluation/how_to_guides/evaluation/llm_as_judge.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+      results = evaluate(
+        dummy_app,
+        data="dataset_name",
+        evaluators=[valid_reasoning]
+      )
+    `,
+    typescript`
+      import type { EvaluationResult } from "langsmith/evaluation";
+      import type { Run, Example } from "langsmith/schemas";
+
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/llm_as_judge.mdx:69]: needs code?

File: docs/evaluation/how_to_guides/evaluation/metric_type.mdx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+:::info Key concepts
+
+- Metrics
+
+:::
+
+LangSmith supports both categorical and numerical metrics, and you can return either when writing a [custom evaluator](../../how_to_guides/evaluation/custom_evaluator).
+
+For an evaluator result to be logged as a numerical metric, it must returned as:
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/metric_type.mdx:17]: this is only true for python, since js can't return just an int, right?

We should be clear about this

File: docs/evaluation/how_to_guides/evaluation/metric_type.mdx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+- an `int`, `float`, or `bool`
+- a dict of the form `{"key": "metric_name", "score": int | float | bool}`
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/metric_type.mdx:20]: nit: a `feedback`[link to reference] dict of the form

File: docs/evaluation/how_to_guides/evaluation/run_evals_api_only.mdx [hunk 11]
```diff
@@ -194,8 +194,8 @@ Next, we'll demonstrate how to run a pairwise experiment. In a pairwise experime
 For more information, check out [this guide](../evaluation/evaluate_pairwise).
 
 ```python
-# A comparative experiment allows you to provide a preferential ranking on the outputs of two or more experiments
-# Spec: https://api.smith.langchain.com/redoc#tag/datasets/operation/create_comparative_experiment_api_v1_datasets_comparative_post
+ A comparative experiment allows you to provide a preferential ranking on the outputs of two or more experiments
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/run_evals_api_only.mdx:197]: meant to be uncommented?

File: docs/evaluation/how_to_guides/evaluation/summary.mdx [hunk 12]
```diff
@@ -0,0 +1,76 @@
+import {
+  CodeTabs,
+  python,
+  typescript,
+} from "@site/src/components/InstructionsWithCode";
+
+# How to run an aggregate evaluation
```
Comment (jakerachleff) [inline docs/evaluation/how_to_guides/evaluation/summary.mdx:7]: Any way to make it clear to users who want to calculate things like f1 score that they should go here from our indexes? I feel like no one ever finds this page
